src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -377,6 +258,220 @@ private void basicAck(Channel channel, long deliveryTag) throws IOException {
         channel.basicAck(deliveryTag, false);
     }
 
+    private Connection createConnection(String connectionName, ConnectionFactory factory, Runnable onFailedRecoveryConnection) {
+        factory.setExceptionHandler(new ExceptionHandler() {
+            @Override
+            public void handleUnexpectedConnectionDriverException(Connection conn, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleReturnListenerException(Channel channel, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConfirmListenerException(Channel channel, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleBlockedListenerException(Connection connection, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConsumerException(Channel channel, Throwable exception, Consumer consumer, String consumerTag, String methodName) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleConnectionRecoveryException(Connection conn, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleChannelRecoveryException(Channel ch, Throwable exception) {
+                turnOffReadiness(exception);
+            }
+
+            @Override
+            public void handleTopologyRecoveryException(Connection conn, Channel ch, TopologyRecoveryException exception) {
+                turnOffReadiness(exception);
+            }
+
+            private void turnOffReadiness(Throwable exception){
+                switch (connectionName) {
```

Comment:
I think there are too many `switch (connectionName)`.
Maybe it's better to add `PublishConnectionManager` and `ConsumeConnectionManager` with `ConnectionManager` as parent (or without parent), encaplulate all logic in them and create them in `ConnectionManager` constructor?

Reply:
Sorry that I confused you, looks like now `PublishConnectionManager` and `ConsumeConnectionManager` are very similar.
I meant that we will not duplicate code in many `switch (connectionName)` and also in these "managers".

Reply:
not actual